docs/references/t-schnorr-how-it-works.mdx

```diff
@@ -0,0 +1,65 @@
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Threshold Schnorr signatures reference
+
+<MarkdownChipRow labels={["Reference"]} />
+
+## Overview
+Currently, this page (only) contains a technical specification for performing Ed25519 hierarchical key derivation.
+This specification is relevant when using the Ed25519 variant of threshold Schnorr signatures on ICP.
+It is, for example, of interest as reference for developers who implement libraries that perform this type of key derivation.
+
+## Ed25519 hierarchical key derivation
+
+This section describes a child key derivation (CKD) function for computing child public keys from parent public keys.
+The section is inspired by [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki) and uses similar wording and structure.
+
+### Motivation
+
+To support the Ed25519 variant of threshold Schnorr signatures on the Internet Computer, a key derivation scheme compatible with Ed25519 signatures is required.
+For a respective signing service on the Internet Computer to be efficient, the signing subnet maintains only a single master key pair and _derives_ signing child keys for each canister.
+Although there exist various hierarchical key derivation schemes (e.g., [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki), [SLIP10](https://github.com/satoshilabs/slips/blob/master/slip-0010.md), [BIP32-Ed25519](https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf), [Schnorrkel](https://github.com/w3f/schnorrkel)), none of the analyzed schemes is suitable for the Internet Computer's (threshold) setting.
+For this reason, a new derivation scheme is specified here.
+The specified scheme does not make use of _clamping_, because it is unnecessary in the given setting, and satisfies the following requirements:
+
+- Off-chain availability: New public keys can be computed off-chain from a master public key without requiring interaction with the IC.
+- Hierarchical derivation: Derived keys are organized in a tree such that from any public key it is possible to derive new child keys. This tree can have arbitrary depth. The first level is used to derive unique canister-specific keys from the master key.
```

Comment:
Actually the tree is limited to 255 levels, matching BIP32

Reply:
There is indeed a restriction to 255 levels. However, this is not a restriction of the key derivation scheme specified in this PR (which scheme stands somewhat by itself), but rather an artificial restriction we impose as per the IC interface spec to protect against various kinds of attacks. In any case, at least for now, I removed the sentence "This tree can have arbitrary depth." to avoid confusion. Let me know if you think I should add back the sentence.

Reply:
Fair - unlike BIP32 there is no external constraint that limits us to depth 255 vs some other depth.